### DISCUSSION TITLE
Errors

### DISCUSSION THREAD

**Discussion by smurraystrata** [Budibase/budibase#9181] [Ideas]
We have a dashboard that makes a few API calls which the interval plugin refreshes every 30 seconds. This works great and looks terrific except for one thing - occasionally (anywhere from 30 minutes to 4 hours) one of the API calls will fail. That's fine, it refreshes again in 30 seconds and doesn't matter, except it pops up a big red error which has to be manually closed by someone. 

My suggestion is to either have an option on a data source to control what it does when there are errors (how long to show it, maybe send it as an email instead of a popup) or not allow public viewers of a dashboard to see the error.

**Comment by melohagan** [comment]
Hey @jedman20 

It's a good idea, although this is not a typical use-case. Usually if the front-end is making an API call you want to user to be alerted if something has gone wrong. 

In future I think there's plans to possibly push notifications from automations. That way you could have an automation with a cron job that can notify users on success/error.

At the moment you could use a Cron automation in combination with the Send Email action if that would be useful to you?

**Reply by smurraystrata** [reply]
Hello @melohagan,

Sorry for being a jerk and not responding. I thought I had. The Cron automation would help if we were trying to monitor for the API calls failing, but doesn't help in this case because we are not concerned about the calls failing, just showing errors on our dashboards. Currently when there is an error the page just shows a blank result for the calls that failed, but it also shows a big red error that has to be dismissed. Where they are dashboards being used by several end users in our company I'd prefer not to have the errors show, or at least have them time out after 30 seconds.

I do wish I could get the third-party API calls not to fail at all, but that is outside my control.

The best case would be that they would only show for certain users, like the ones with the ability to edit the dashboard.

Thank you!
